packages/frontend/src/pages/UserPage.jsx

```diff
@@ -26,7 +27,7 @@ export default observer(() => {
             </div>
             <div className='flex'>
               <h5>Attesters Joined</h5>
-              <h6>2</h6>
+              {/* <h6>{user.signUpsByUser.length}</h6> */}
```

Comment:
Yes, this is related to the Home page bug above. 

A react component (nowadays) is usually a single function. This function is called any time the component is re-rendered. This means if I have:

```js
export default () => {
  console.log('hello')
  return (
    <div>Hello</div>
  )
}
```

`hello` will be printed each time the component is rendered. With mobx components are automatically re-rendered anytime changes are made to data the component is observing.

On the home page some data is being changed once the page is rendered, which is causing a second render. On this page (`userPage`) the component calls a function that asynchronously changes `user.signUpsByUser.length`. This async change will cause the page to re-render, which will cause another update, which will cause another change, etc.

The way to avoid this is [`useEffect`](https://www.w3schools.com/react/react_useeffect.asp). This will cause react to run some code on first render, and any time observed variables (a parameter of `useEffect`) change.

So to fix the example above:

```js
export default () => {

  React.useEffect(
    () => console.log('hello'), //~ function to run
    [] //~ variables to observe (none)
  );


  return (
    <div>Hello</div>
  )
}
```

Now `hello` will be printed once in the component lifecycle. Note that if you have multiple copies of this component, each will print `hello` once.

Also note that I'm passing a second argument `[]` to `useEffect`. This causes the function to be called only on first render. `[]` is an empty list of variables to observe.